pkg/config/configobserver/configobserver_test.go

```diff
@@ -0,0 +1,146 @@
+package configobserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	clsetfake "k8s.io/client-go/kubernetes/fake"
+	corefake "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	"k8s.io/klog"
+
+	clienttesting "k8s.io/client-go/testing"
+)
+
+func TestChangeSupportConfig(t *testing.T) {
+	var cases = []struct {
```

Comment:
I would rather create 2 tests, each test with only one purpose. It's more clear to read and understand. But up to you.